### DISCUSSION TITLE
[❄️ Snowflake Official] Old Grant Resources Deprecation Notice

### DISCUSSION THREAD

**Discussion by sfc-gh-swinkler** [Snowflake-Labs/terraform-provider-snowflake#1959] [General]
TL;DR: While we have marked the older grant resources as deprecated, we are not removing them from the provider for several months.

Per [Grant Issues Update](https://github.com/Snowflake-Labs/terraform-provider-snowflake/discussions/1890) and [Grant Resource Refactoring](https://github.com/Snowflake-Labs/terraform-provider-snowflake/discussions/1470) we have marked all of the older grant resources as deprecated in favor of new grant resources. As discussed in those posts, the older grant resource code is fragile, and maintenance offers significant regression risks.

Specifically:

- `snowflake_grant_priviliges_to_role` (implemented)
- `snowflake_grant_privileges_to_database_role` (not yet implemented)
- `snowflake_grant_privileges_to_share` (not yet implemented)
- `snowflake_grant_privileges_to_application_role` (not yet implemented)

We will likely remove the existing resources from the provider five months from now, at the end of 2023. There is plenty of time to adopt the new resources, and you do not have to migrate immediately. Due to a many-to-one resource consolidation, migrating some existing grants into the new format may take time and effort. We recognize the challenges and plan to create documentation on the import procedure. Please share any barriers or concerns you have about this migration here.

**Comment by chrisweis** [comment]
@sfc-gh-swinkler, Terraform n00b here, doing my first deployment with this mostly awesome provider (thanks!).  If I'm running v0.66.1, does this mean that:
1. Version 0.66.1 will begin **failing** at the end of 2023, or
2. I just can't continue to **upgrade** my TF code to the latest version (currently v0.69.0) and have it work (with "Warning: Deprecated Resource" messages)?

Thanks

**Comment by cpmub1** [comment]
@sfc-gh-swinkler  i have tried to migrate to the new resource. But, as such modification is dropping the old resource & adding the new resource. So, it seems, we have to import the new resource & delete the old resource from the state file - individually. is there any option to do mass conversion or adoption of new resource at state file level ?
Its too cumbersome to migrate fine grained controls. for eg. we control the privileges at schema level - and for about 10 databases, i have 1800+ resources that needs to be migrated to new resource. manual import & delete is error prone.

ie. documentation is great, but is it possible to imbibe any automation in converting the old resource type to new resource type ?

**Comment by ugmuka** [comment]
Hi @sfc-gh-swinkler,

When do you plan to publish documentation on the import procedure? We are in the process of planning a migration for our grant resources.

**Comment by kanomaxb** [comment]
Hi @sfc-gh-swinkler,

Have you given up the `grant_ownership_to_role` mentioned in #1890 ? Should we use `grant_priviliges_to_role` for ownership transfer?

**Comment by sfc-gh-jcieslak** [comment]
Closing as the grant redesign was concluded. More info [here](https://github.com/Snowflake-Labs/terraform-provider-snowflake/discussions/1890#discussioncomment-9071073).
